apps/frontend/src/app/auth/admin/admin.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { ServerService } from '../../shared/server.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AdminService {
+  isAdmin: boolean | null = null;
+
+  constructor(private serverSrv: ServerService, private router: Router) {}
+
+  autoAdmin() {
+    return this.isAdmin;
+  }
+
+  getIsAdmin() {
+    if (this.isAdmin === null) {
+      this.requestIsAdmin();
+      return this.isAdmin;
+    } else {
+      return this.isAdmin;
+    }
+  }
+
+  requestIsAdmin() {
+    this.serverSrv.isAdmin().subscribe({
+      next: (result) => {
+        this.isAdmin = result.isAdmin;
+      },
+      error: () => {
+        this.isAdmin = false;
+      },
+    });
+  }
```

Comment:
subscribe bitte nicht im Service sondern in der Komponente und danach unsubscribe diesen Subscriber. sonst wird dieser Subscriber nie destroyed.
wenn man den Wert des Subscribers im Service braucht, muss man mit der Map-Function aus rxjs library arbeiten